tests/app.test.ts

```diff
@@ -1,8 +1,99 @@
 import request from "supertest";
-import  app from "../source/app";
+import router from '../routes/routes';
+import { Soldier } from "../repository/repo";
+const app = router; 
+beforeEach(async ()=>{ 
+    await Soldier.deleteMany();
+});
 
 describe('Is Soldier Alive', () => {
     it('Should Send an API Get Request If Soldier Is Alive, Expected 200:', () => {
         request(app).get('/health').expect("I am alive!");
     });
 });
+
+describe('Insert Soldier into DB #important: you need to modify the ID of inserted soldier in the test file in order to succeed this test', () => {
+    it('Should Send an API Post Request and save soldier to DB, Expected 201:',async () => {
+
+        const SoldierToBeAdded = {
+            "_id": "179",
+            "name": "cringus",
+            "rank":"Silver", 
+            "limitations": [
+              "Useless",
+              "no_one_likes_this_guy"
+            ]
+          }
+        await request(app).post('/soldiers').send(SoldierToBeAdded).expect(201);
+    });
+});
+
+describe('Fail to insert Soldier into DB', () => {
+    it('Should Send an API Post Request and not save soldier to DB, Expected 400:',async () => {
+        const SoldierToBeAdded = {
+            "_id": "111",
+            "name": "cringus",
+            "limitations": [
+              "Useless",
+              "no_one_likes_this_guy"
+            ],
+          }
+        await request(app).post('/soldiers').send(SoldierToBeAdded).expect(400);
+
+
+    });
+});
+describe('Is Soldier in DB?', () => {
+    it('Should Send an API Get Request If Soldier Is In DB, Expected 200:', async () => {
+    
+        const testSoldier = {
+            "_id": "1",
+            "name": "Test Soldier",
+            "rank":"Test boi", 
+            "limitations": [
+              "Useless",
+              "no one likes this guy"
+            ],
+          }
+        const sentSoldier = new Soldier(testSoldier);  await sentSoldier.save();
+        console.log(testSoldier); 
+        request(app).get('/soldiers/1').expect(200); 
```

Comment:
again, you need to test for what you want, not just if the response is in the correct code